File: instantlib/src/main/java/com/baselib/instant/thread/IThreadPoolManager.java
package com.baselib.instant.thread;

/**
 * 线程池管理对象需实现的接口
 * <p>
 * 指定操作
 *
 * @author wsb
 */
public interface IThreadPoolManager {

    /**
     * 执行任务
     *
     * @param task 目标task
     */
    void execute(Runnable task);

    /**
     * 执行取消
     *
     * @param task 目标task
     */
    void cancel(Runnable task);

    /**
     * 资源回收
     * */
    void cleanUp();
}

File: instantlib/src/main/java/com/baselib/instant/executor/IThreadPoolManager.kt
package com.baselib.instant.executor

/**
 * 线程池管理对象需实现的接口
 *
 * 指定操作
 *
 * @author wsb
 */
interface IThreadPoolManager {
    /**
     * 执行任务
     *
     * @param task 目标task
     */
    fun execute(task: Runnable)

    /**
     * 执行取消
     *
     * @param task 目标task
     */
    fun cancel(task: Runnable)

    /**
     * 资源回收
     */
    fun cleanUp()
}